### PR TITLE
[BEAM-1151] Add interface for accessing failed BigQuery inserts

### PR DESCRIPTION
**Pull request by reuvenlax**
This PR adds a new return type to BigQueryIO.Write, to replace PDone. This enables future features such as accessing failed inserts as a PCollection. 

R: @dhalperi

### CODE REVIEW

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  /**
+   * Returns a {@link PCollection} of inserts that permanently failed when attempting to insert into
+   * BigQuery. This can be used for example to write these rows out to an alternate sink. This wil
+   * only be set if the user has specified a failed-row policy on {@link BigQueryIO.Write}.
+   *
+   * <p>This functionality is not yet implemented.
+   */
+  public PCollection<TableRow> getFailedRows() {
```
Comment (dhalperi) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.java]: drop from this PR

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  /**
+   * Creates a {@link WriteResult} in the given {@link Pipeline}.
+   */
+  public static WriteResult in(Pipeline pipeline) {
+    return new WriteResult(pipeline);
+  }
+
+  @Override
+  public List<TaggedPValue> expand() {
+    //
```
Comment (dhalperi) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.java]: ?

Comment (reuvenlax) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.java]: done

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.java (hunk 1 above)
Comment (reuvenlax) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.java]: done

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+
+/**
+ * The result of a {@link BigQueryIO.Write} transform.
+ */
+public class WriteResult extends POutputValueBase {
+  /**
+   * Creates a {@link WriteResult} in the given {@link Pipeline}.
+   */
+  public static WriteResult in(Pipeline pipeline) {
```
Comment (dhalperi) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.java]: should this be a package-private constructor? Do we think users outside this module would need to create them?

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.values.POutputValueBase;
+import org.apache.beam.sdk.values.TaggedPValue;
+
+
+/**
+ * The result of a {@link BigQueryIO.Write} transform.
+ */
+public class WriteResult extends POutputValueBase {
```
Comment (dhalperi) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.java]: `final` class?

Comment (dhalperi) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.java]: Or if we think users will want to mock for testing, interface with private default implementation

Comment (reuvenlax) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.java]: done

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.java (hunk 3 above)
Comment (reuvenlax) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.java]: done

**Pull request closed without merging**